cmd/tealdbg/cdtState.go

```diff
@@ -704,21 +839,25 @@ func makeGlobals(s *cdtState, preview bool) (desc []cdt.RuntimePropertyDescripto
 	fields := prepareGlobals(s.globals)
 	desc = make([]cdt.RuntimePropertyDescriptor, len(fields))
 	for i, field := range fields {
+		// We need to dynamically override this field with the proper value each step.
+		if field.Name == "OpcodeBudget" {
```

Comment:
I think `state.Update()` should be updating globals value. This is only "renderer" and should not contains any mapping/fields updating logic.

Reply:
I changed the `globals[opcodeBudgetFieldIdx]` instead :+1: